### PR TITLE
take partition size into account on checking replication delay

### PR DESCRIPTION
**Pull request by farzadsedghi**
The check for replication delay should compare the last replicated timestamp with the end of the partition

### CODE REVIEW

File: dbeam/src/main/scala/com/spotify/dbeam/PsqlAvroJob.scala [hunk 1]
```diff
@@ -65,21 +65,26 @@ object PsqlAvroJob {
...
     validateOptions(args)
 
-    validateReplication(args.partition.get, queryReplication(args.createConnection()))
+    val partition = args.partition.get
+    val lastReplication = queryReplication(args.createConnection())
+    val partitionPeriod: ReadablePeriod = args.partitionPeriod
+
+    if (isReplicationDelayed(partition, lastReplication, partitionPeriod)) {
+      log.error(s"Replication was not completed for partition, " +
+        s"expected >= $partition, actual = $lastReplication")
```
Comment (blazy2k9) [inline dbeam/src/main/scala/com/spotify/dbeam/PsqlAvroJob.scala]: it should be `partition.plus(partitionPeriod)`

Comment (o0neup) [reply dbeam/src/main/scala/com/spotify/dbeam/PsqlAvroJob.scala]: And after having a second look I wonder do we really need the `isReplicationDelayed` method? It  's supposed to hide the validation logic but we exposing it in the log message anyway. 
The alternative would be to put the log msg back into method. What do you think?

File: dbeam/src/test/scala/com/spotify/dbeam/PsqlAvroJobTest.scala [hunk 2]
```diff
@@ -85,19 +85,17 @@ class PsqlAvroJobTest extends FlatSpec with Matchers with BeforeAndAfterAll {
...
     val partition = new DateTime(2027, 7, 31, 0, 0, DateTimeZone.UTC)
-    val lastReplication = new DateTime(2027, 7, 19, 0, 0, DateTimeZone.UTC)
+    val lastReplication = new DateTime(2027, 7, 31, 10, 0, DateTimeZone.UTC)
+    val partitionPeriod = Days.ONE
 
-    a[IllegalArgumentException] should be thrownBy {
-      PsqlAvroJob.validateReplication(partition, lastReplication)
-    }
+    PsqlAvroJob.isReplicationDelayed(partition, lastReplication, partitionPeriod) shouldBe true
   }
```
Comment (labianchin) [inline dbeam/src/test/scala/com/spotify/dbeam/PsqlAvroJobTest.scala]: I think we can have a few more tests here, to validate the boundaries.

Like:
- `x<y`
- `x=y`
- `x>y`

And maybe also:
- `x-1<y`
- `x<y+1`

File: dbeam/src/main/scala/com/spotify/dbeam/PsqlAvroJob.scala (hunk 1 above)
Comment (blazy2k9) [reply dbeam/src/main/scala/com/spotify/dbeam/PsqlAvroJob.scala]: I agree. I think it’s safe to move the error log to the `isReplicationDelayed` method.

**Comment by farzadsedghi** [comment]
@labianchin updated and also rebased to scio 0.4.7 upgrade PR

File: dbeam/src/test/scala/com/spotify/dbeam/PsqlAvroJobTest.scala [hunk 3]
```diff
@@ -82,22 +82,45 @@ class PsqlAvroJobTest extends FlatSpec with Matchers with BeforeAndAfterAll {
     PsqlAvroJob.validateOptions(options)
   }
 
-  it should "validate replication" in {
+  it should "valid replication state, replicated until end of partition" in {
```
Comment (labianchin) [inline dbeam/src/test/scala/com/spotify/dbeam/PsqlAvroJobTest.scala]: "it should validate" instead?

Comment (labianchin) [reply dbeam/src/test/scala/com/spotify/dbeam/PsqlAvroJobTest.scala]: or succeed? (opposite of fail)

File: dbeam/src/test/scala/com/spotify/dbeam/PsqlAvroJobTest.scala [hunk 4]
```diff
@@ -82,22 +82,45 @@ class PsqlAvroJobTest extends FlatSpec with Matchers with BeforeAndAfterAll {
...
     val partition = new DateTime(2027, 7, 31, 0, 0, DateTimeZone.UTC)
     val lastReplication = new DateTime(2027, 8, 1, 0, 0, DateTimeZone.UTC)
+    val partitionPeriod = Days.ONE
 
-    val actual = PsqlAvroJob.validateReplication(partition, lastReplication)
+    PsqlAvroJob.isReplicationDelayed(partition, lastReplication, partitionPeriod) shouldBe false
+  }
+
+  it should "valid replication state, replicated until the next day" in {
```
Comment (labianchin) [inline dbeam/src/test/scala/com/spotify/dbeam/PsqlAvroJobTest.scala]: same as above
